sdk-extensions/autoconfigure-spi/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/IterableConfigProperties.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.autoconfigure.spi;
+
+import io.opentelemetry.api.internal.ConfigUtil;
+import java.util.Iterator;
+import java.util.Set;
+
+/** Config properties that allow access to the property names. */
+public interface IterableConfigProperties extends ConfigProperties, Iterable<String> {
```

Comment:
Why introduce a subtype instead of extending `ConfigProperties`? Will `ConfigProperties` ever be passed as an argument once `IterableConfigProperties` is available?